examples/dev/triangle_edge.py

```diff
@@ -0,0 +1,379 @@
+    vec[:, 0] = path_
+    vec[:-1, 1] = directions
+    if closed:
+        # point from the last vertex towards the first vertex
+        vec[-1, 1] = (
+                (path_[0] - path_[-1]) / np.linalg.norm(path_[-1] - path_[0])
+                )
+    else:
+        # point back towards the penultimate vertex
+        vec[-1, 1] = -vec[-2, 1]
+    return vec
+
+
+def generate_miter_helper_vectors(
+        direction_vectors_array: np.ndarray
+        ) -> np.ndarray:
+    """Generate the miter helper vectors.
+
+    The miter helper vectors are a pair of vectors for each point in the path,
+    which together help define whether a bevel join will be needed. The first
+    vector is half of the normalized direction vector for that vertex, and the
+    second is *minus* half of the normalized direction vector for the previous
+    vertex. Their angle is the angle of the edges at that vertex.
+
+    Parameters
+    ----------
+    direction_vectors_array : array of shape (n, 2)
+        array of normalized (length 1) direction vectors for each point in the
+        path.
+
+    Returns
+    -------
+    array of shape (n, 2, 2)
+        array of miter helper vectors
+    """
+    half_direction = direction_vectors_array.copy()
+    half_direction[:, 1] *= 0.5
+    half_prev_direction = half_direction.copy()
+    half_prev_direction[:, 1] *= -1
+    half_prev_direction[:, 1] = np.roll(half_prev_direction[:, 1], 1, axis=0)
+    half_prev_direction[:, 0] += half_direction[:, 1]
+    return np.concatenate([half_direction, half_prev_direction], axis=0)
+
+
+@numba.njit
+def generate_orthogonal_vectors(direction_vectors: np.ndarray) -> np.ndarray:
+    """Generate the orthogonal vectors to the direction vectors.
+
+    The orthogonal vector starts at the middle of the direction vector and is
+    orthogonal to it in the positive orientation. Its length is half of the
+    direction vector, to align with the normalized edge width.
+
+    Parameters
+    ----------
+    direction_vectors : array, shape (n, 2, 2)
+        The direction vector start points (``direction_vectors[:, 0, :]``) and
+        directions (``direction_vectors[:, 1, :]``).
+
+    Returns
+    -------
+    orthogonal_vectors : array, shape(n, 2, 2)
+        The orthogonal vector start points and directions.
+    """
+    position = 0
+    vector = 1
+    y, x = 0, 1
+    half_direction = 0.5 * direction_vectors[:, 1, :]
+    orthogonal_vectors = direction_vectors.copy()
+    orthogonal_vectors[:, position] = (
+            direction_vectors[:, position] + half_direction
+            )
+
+    orthogonal_vectors[:, vector, y] = -half_direction[:, x]
+    orthogonal_vectors[:, vector, x] = half_direction[:, y]
+    return orthogonal_vectors
+
+
+@numba.njit
+def generate_miter_vectors(
+        mesh: tuple[np.ndarray, np.ndarray, np.ndarray]
+        ) -> np.ndarray:
+    """For each point on path, generate the vectors pointing to miter points.
```

Comment:
It is not fully true. It also works with bevel join. 
So maybe we should update the function name and doc string. 

But I do not know how to name these points that are used to create triangles and lying outside path.

Reply:
I'm ok calling them "miter points" for now — I certainly do so in my head now after so much time here! 😂 We just happen to use one of the miter points in the bevel join! 😉